### PR TITLE
[Bindless][Exp] Add device queries for sampled image fetch

### PR DESCRIPTION
**Pull request by przemektmalon**
Added the following queries for device capabilities of fetching sampled images:

- DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_1D_USM_EXP
- DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_1D_EXP
- DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_2D_USM_EXP
- DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_2D_EXP
- DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_3D_USM_EXP
- DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_3D_EXP

### CODE REVIEW

**Comment by przemektmalon** [comment]
Related DPC++ MR: https://github.com/intel/llvm/pull/12447

**Review by isaacault — APPROVED** [review]

File: scripts/core/exp-bindless-images.yml [hunk 1]
```diff
@@ -86,6 +86,24 @@ etors:
     - name: INTEROP_SEMAPHORE_EXPORT_SUPPORT_EXP
       value: "0x200F"
       desc: "[$x_bool_t] returns true if the device supports exporting internal event resources"
+    - name: BINDLESS_SAMPLED_IMAGE_FETCH_1D_USM
```
Comment (fabiomestre) [inline scripts/core/exp-bindless-images.yml:89]: Since bindless is still experimental, I'm wondering if these enums should have the `EXP` suffix?

Comment (przemektmalon) [reply scripts/core/exp-bindless-images.yml]: Good catch, I missed that. Updated to have `_EXP` suffix.

**Review by fabiomestre — APPROVED** [review]

**Comment by przemektmalon** [comment]
Friendly ping @oneapi-src/unified-runtime-cuda-write

**Comment by przemektmalon** [comment]
Friendly ping to @oneapi-src/unified-runtime-cuda-write and @ldrumm for re-reviews/approvals/merging.

The changes in this MR are minimal, however I would need them to be merged before I can update the related [DPC++ MR](https://github.com/intel/llvm/pull/12447) to make it ready to review and get it closer to merging.

Note that the related [DPC++ MR](https://github.com/intel/llvm/pull/12447) fails CI but this is stemming from tests that are unrelated to that particular MR/

**Review by npmiller — APPROVED** [review]
CUDA changes LGTM

**Review by DBDuncan — APPROVED** [review]

**Comment by przemektmalon** [comment]
Marked as ready to merge. Corresponding [DPC++ PR](https://github.com/intel/llvm/pull/12447) has gotten approvals and passed the last CI (currently running again due to git merge, but it is passing).

**Pull request merged**